### PR TITLE
feat(credentials): added credential sendProblemReport method

### PR DESCRIPTION
**Pull request by Iskander508**
Signed-off-by: Pavel Zarecky <zarecky@procivis.ch>

Adding a convenience method to send problem report as part of RFC's 0036 and 0453 unhappy flow.
It is a complementary implementation to the `agent.proofs.sendProblemReport`. But for credentials it was still missing.

### CODE REVIEW

File: packages/core/src/modules/credentials/CredentialsModule.ts
Comment (TimoGlastra) [inline packages/core/src/modules/credentials/CredentialsModule.ts:70]: Maybe we can also make this an object like we do with the other methods that contain both the `credentialRecordId` and `message`? Mainly for two reasons:
1. Consistency with other methods
2. It would allow to add more options in the future without needing to make a breaking change (we'll probably want to make it an options object when adding more options)

File: packages/core/src/modules/credentials/services/CredentialService.ts
Comment (TimoGlastra) [inline packages/core/src/modules/credentials/services/CredentialService.ts:88]: Same here, if we make it an object it would be easier to add new properties later on (such as exposing the `code`)

**Review by TimoGlastra — COMMENTED** [review]
This is great @Iskander508!

File: packages/core/src/modules/credentials/CredentialsModule.ts
Comment (Iskander508) [reply packages/core/src/modules/credentials/CredentialsModule.ts]: @TimoGlastra refactored

File: packages/core/src/modules/credentials/services/CredentialService.ts
Comment (Iskander508) [reply packages/core/src/modules/credentials/services/CredentialService.ts]: refactored

**Review by TimoGlastra — APPROVED** [review]
🚀

**Comment by TimoGlastra** [comment]
@Iskander508 I'll merge this once #908 is merged

**Pull request merged**
